src/components/signup/SignUp.js

```diff
@@ -1,25 +1,93 @@
 import "./SignUp.css";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { useAuth } from "../../contexts/auth-context";
 
 const SignUp = () => {
+  const { authState, authDispatch } = useAuth();
+
+  const [userInfo, setUserInfo] = useState({
+    firstname: "",
+    secondname: "",
+    mobile: "",
+    email: "",
+    password: "",
+  });
+
+  const inputHandler = (e) => {
+    e.preventDefault();
+
+    setUserInfo((data) => ({ ...data, [e.target.name]: e.target.value }));
+  };
+  const postSignUpDetails = async (e) => {
+    e.preventDefault();
+    authDispatch({ type: "USER_LOAD" });
+
+    try {
+      const { data } = await axios.post("/api/auth/signup", userInfo);
+      authDispatch({ type: "USER_LOAD_SUCCESS", payload: data.createdUser });
+      console.log(data);
+      localStorage.setItem("token", data.encodedToken);
+    } catch (err) {
+      console.log(err);
+    }
+    // console.log(authState);
+  };
+
+  useEffect(() => {
+    console.log(authState);
+  });
```

Comment:
why are you using this useEffect just to console.log ? , you can remove this

Reply:
it was just testing purpose , now I will remove it .